client/app/components/QueryEditor.jsx

```diff
@@ -81,92 +63,125 @@ class QueryEditor extends React.Component {
 
   constructor(props) {
     super(props);
+
     this.state = {
       schema: null, // eslint-disable-line react/no-unused-state
-      keywords: [], // eslint-disable-line react/no-unused-state
+      keywords: {
+        table: [],
+        column: [],
+        tableColumn: [],
+      },
       autocompleteQuery: localOptions.get('liveAutocomplete', true),
       liveAutocompleteDisabled: false,
       // XXX temporary while interfacing with angular
       queryText: props.queryText,
     };
-    langTools.addCompleter({
+
+    const schemaCompleter = {
+      identifierRegexps: [/[a-zA-Z_0-9.\-\u00A2-\uFFFF]/],
       getCompletions: (state, session, pos, prefix, callback) => {
-        if (prefix.length === 0) {
+        const tableKeywords = this.state.keywords.table;
+        const columnKeywords = this.state.keywords.column;
+        const tableColumnKeywords = this.state.keywords.tableColumn;
+
+        if (prefix.length === 0 || tableKeywords.length === 0) {
           callback(null, []);
           return;
         }
-        callback(null, this.state.keywords);
-      },
-    });
 
-    this.onLoad = (editor) => {
```

Comment:
Was there a reason for those methods to be inside QueryEditor's constructor?

Reply:
🤷‍♂️ 

@washort ?